projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/ExecutionDataFetcher.java

```diff
@@ -305,4 +286,52 @@ private static DataJobPage buildResponse(
             .totalPages(pageSize)
             .totalItems(count).build();
    }
+
+   List<V2DataJob> populateStatusCounts(List<V2DataJob> dataJos, DataFetchingEnvironment dataFetchingEnvironment) {
+
+      var statusesToCount = determineStatusesToCount(dataFetchingEnvironment);
+      var jobsList = dataJos.stream().map(V2DataJob::getJobName).collect(Collectors.toList());
+      var response = jobExecutionService.countExecutionStatuses(jobsList, statusesToCount);
+
+      dataJos.stream()
+              .forEach(job -> {
+                 if (job.getDeployments() != null) {
+                    job.getDeployments()
+                            .stream()
+                            .findFirst()
```

Comment:
is findFirst necessary?

Reply:
I'm following what already implemented logic does here: https://github.com/vmware/versatile-data-kit/blob/563c92cf79936f99b006f24e17ea0b85fd11c5c0/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/ExecutionDataFetcher.java#L66 

We expect to have a single deployment, so it's a choice between findAny() and findFirst() which achieve the same thing I guess.

Reply:
```
dataJob.getDeployments()
                    .stream()
                    .forEach(deployment -> deployment.setExecutions(
                            jobsExecutionRepository.findDataJobExecutionsByDataJobName(dataJob.getJobName(), pageable)
                                    .stream()
                                    .map(ToApiModelConverter::jobExecutionToConvert)
                                    .collect(Collectors.toList())));
```

That's a bit simpler .. But leave it as it is . It's better to be consistent for now.